### PR TITLE
Local "push" item implemented, handling ConnectionError added

### PR DESCRIPTION
**Pull request by panasevychol**
Local "push" item implemented, handling ConnectionError while request added

## Proposed changes

Local "push" item implemented, handling ConnectionError while request added

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../master/CONTRIBUTING.rst) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/cli/registry/push.py [hunk 1]
```diff
@@ -93,3 +95,59 @@ def push_item(item_type: str, item_name: str) -> None:
...
+
+def get_packages_path() -> str:
+    """
+    Get path to packages folder.
+
+    Should not be called from outside the project dir.
+
+    :return: str path to packages dir.
+    """
+    project_parent_dir = os.path.abspath('__file__').split(
```
Comment (DavidMinarsch) [inline aea/cli/registry/push.py]: We have a path already defined as a constant: `DEFAULT_REGISTRY_PATH`. Use this instead.

File: tests/test_cli/test_registry/test_push.py [hunk 2]
```diff
@@ -101,3 +108,97 @@ def test_remove_pycache_no_pycache(self, path_exists_mock, rmtree_mock):
         source_dir = 'somedir'
         _remove_pycache(source_dir)
         rmtree_mock.assert_not_called()
+
+
+@mock.patch(
+    'aea.cli.registry.push.os.path.abspath',
+    return_value=('somepath/agents-aea/somefile')
```
Comment (DavidMinarsch) [inline tests/test_cli/test_registry/test_push.py]: see above, should reference DEFAULT_REGISTRY_PATH not agents-aea

File: aea/cli/registry/push.py (hunk 1 above)
Comment (panasevychol) [reply aea/cli/registry/push.py]: I've tried. But it always leads to "packages" one level up. It won't work with any nested folder structure.
Sure I can use that. But don't you think it's better to have a possibility to use it with nested structure?

Comment (DavidMinarsch) [reply aea/cli/registry/push.py]: well you neet to construct the correct path I suppose. `packages` is assumed to be on the root dir. So if you build cwd + DEFAULT_REGISTRY+PATH then cwd would have to be the agent project dir. You can figure out whether you're in agent project dir by checking if the aea-config file is present.

**Pull request merged**